### ISSUE TITLE
[BUG] Hardcoded Endpoint for OpenAI

### ISSUE THREAD

**Issue by victor-v-rad** [Azure/azure-sdk-for-net#37794]
### Library name and version

Azure.AI.OpenAI_1.0.0-beta.6

### Describe the bug

Hi Team!

Companies create internal API proxies to avoid direct calls to https://api.openai.com/ . But  Azure has OpenAI endpoint hardcoded https://github.com/Azure/azure-sdk-for-net/blob/c8b27f9fca17384a481a1debf565052ffc97b2b9/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L28
without possibility to specify custom endpoint. Constructors use only hardcoded PublicOpenAIEndpoint property
https://github.com/Azure/azure-sdk-for-net/blob/c8b27f9fca17384a481a1debf565052ffc97b2b9/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L120-L131 

### Expected behavior

It should be possible to specify a custom endpoint. Maybe add one more constructor with _nonAzureOpenAIApiKey = openAIApiKey; code. See
https://github.com/Azure/azure-sdk-for-net/blob/c8b27f9fca17384a481a1debf565052ffc97b2b9/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L120-L131

### Actual behavior

Constructors use only hardcoded PublicOpenAIEndpoint property
https://github.com/Azure/azure-sdk-for-net/blob/c8b27f9fca17384a481a1debf565052ffc97b2b9/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L120-L131

### Reproduction Steps

```
OpenAIClient client = new OpenAIClient("your-api-key-from-platform.openai.com");

Response<Completions> response = await client.GetCompletionsAsync(
    "text-davinci-003", // assumes a matching model deployment or model name
    "Hello, world!");

foreach (Choice choice in response.Value.Choices)
{
    Console.WriteLine(choice.Text);
}
```

**Comment by gingters** [comment]
Just to second that, if you want to use i.e. Semantic Kernel to enrich your Blazor WebAssembly application with AI capabilities, you for sure don't want to provide your API key to the browser ;)

It's easy to add a Yarp.ReverseProxy to your server that authenticatces your users based on your normal auth mode (Cookies, Tokens etc.) and then redirects all calls to /openai to your Azure OpenAI endpoints and add the secret Api Key to these requests on the server side, but since you can't set the base url for the pure OpenAI endpoint it's sadly not possible to use these from your web application atm.

**Comment by trrwilson** [comment]
Thank you for getting in touch!

There's a generalized mechanism for Azure SDK libraries to configure proxy server use that `Azure.AI.OpenAI` shares as part of the SDK family; you can see the documentation and an example here:

- https://learn.microsoft.com/en-us/dotnet/azure/sdk/azure-sdk-configure-proxy
- https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Configuration.md#configuring-a-proxy

When enterprise and other secured networks limit access to external service endpoints via such a mechanism, this proxy configuration is the preferred way to enable proxied access.

Is this proxy configuration mechanism a viable path for your use case? If it isn't, I'd like to ensure we forward and address the gap holistically, as this situation with operations using static URL bases is shared across many of the Azure SDK libraries.

Thanks again!

**Comment by gingters** [comment]
Hi,

no, that sadly is not sufficient for our use case.
We use Semantic Kernel, building upon this SDK, in a Blazor WebAssembly application.

With SK and this SDK running directly in the Browser the API Key to Access OpenAI would be unprotected in the open. The OpenAI endpoint should be accessed through the server hosting the web application which can inject an API key. As such we would need to set the Base URL to OpenAI to something like "./api/openai".

It is not possible to create a "system wide" proxy configuration in the Browser. So we need a way to override the Base URI.

**Comment by victor-v-rad** [comment]
@trrwilson Thank you for sharing the proxy configuration. It works for me.

**Comment by trrwilson** [comment]
@gingters -- fully understood about environment-variable-based configuration not being viable.

From that [second link I shared](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Configuration.md#configuring-a-proxy), there's also a programmatic mechanism to configure a proxy server via client options; pulling that page together with the OpenAI code, it'd be along the lines of:

```csharp
var httpClientHandler = new HttpClientHandler()
{
    Proxy = new WebProxy("http://my-enterprise.proxy-server.com"),
};
var httpClientWithProxy = new HttpClient(httpClientHandler);
var openAIClientOptions = new OpenAIClientOptions()
{
    Transport = new HttpClientTransport(httpClientWithProxy),
};
var client = new OpenAIClient(
    endpoint: new Uri("https://destination-aoai-resource.openai.azure.com"),
    new AzureKeyCredential("fake-key-to-be-replaced-by-proxy"),
    openAIClientOptions);
```

This will configure the `OpenAIClient` instance to use the underlying `HttpClient` provided, which has in turn been configured (in code, sans environment variable involvement) to use a proxy.

I've scheduled some work for us to add better documentation and test automation coverage to clarify this, but in the interim I've validated by hand that this does indeed invoke a proxy request.
